### DISCUSSION TITLE
Allow disabling polling for YAML entities

### DISCUSSION THREAD

**Discussion by koying** [home-assistant/architecture#664] [Entity Models] [answered]
This is a follow-up to https://github.com/home-assistant/core/pull/56323, where I dropped the ball but I again face a situation where it would be useful.

The basic idea is to mimic the possibility of disabling polling that exists for config_flow integrations since 2021.6, but for YAML entities like command_line, REST, ...
The last suggestion in the PR was to allow an explicit "never" to the "scan_interval" of polling entities, but a (probably better) alternative would be to expose "should_poll".

I already have some code for "never", but I find it cumbersome, specifically because it forces to union timedelta and string for the scan_interval.
If there is agreement this is desirable, I'd go for ```polling: false``` or similar

**Comment by frenck** [comment, marked as answer]
In all honesty... Integrations should be migrated to config entries. That resolves it.
I don't think we should enhance older constructs. As a matter of fact, changing the YAML configuration of integrations that connect to devices or services isn't allowed (as per ADR-0007 and ADR-0010).

I agree on things like REST, but that should be solved in a way we did with the template now, allowing to specify triggers. This is another discussion that is currently going on, on how we can re-use and re-deploy these things to integrations like REST.

**Reply by koying** [reply]
> Integrations should be migrated to config entries. That resolves it.

Yeah, it goes without saying that this is about the "technical" integrations (command-line, REST, ...) that will very likely never migrate to config_flow.

Using triggers is surely an interesting idea, but then you'd have a different way to achieve non-polling depending on whether the integration has config_flow or not, i.e. `homeassistant.update_entity` or not, which, imo, would be quite confusing for the average user (and whoever peruse the forums knows they are already quite confused ;) )
